app/services/flow/flows/provider_start.rb

```diff
@@ -61,17 +61,13 @@ class ProviderStart < FlowSteps # rubocop:disable Metrics/ClassLength
         },
         check_provider_answers: {
           path: ->(application) { urls.providers_legal_aid_application_check_provider_answers_path(application) },
-          forward: ->(application) do
-            if Setting.override_dwp_results?
-              application.non_passported? ? :confirm_dwp_non_passported_applications : :check_benefits
-            else
-              :check_benefits
-            end
-          end
+          forward: ->(_application) { :check_benefits }
         },
         check_benefits: {
           path: ->(application) { urls.providers_legal_aid_application_check_benefits_path(application) },
-          forward: ->(application) do
+          forward: ->(application, confirm_non_passported) do
+            return :confirm_dwp_non_passported_applications if confirm_non_passported
+
```

Comment:
Below this on lines 71-81, I think this can be refactored
```

            if application.applicant_receives_benefit?
              application.change_state_machine_type('PassportedStateMachine')
              application.used_delegated_functions? :substantive_applications : :capital_introductions
            else
              application.change_state_machine_type('NonPassportedStateMachine')
              :confirm_dwp_non_passported_applications ? :confirm_dwp_non_passported_applications : :applicant_employed
            end
            
            
```

Reply:
Fair. It's actually true here if the feature flag is enabled _and_ the application is `non_passported`, so I was trying to give it a name that wasn't purely based around the flag. But you're right that it could be confusing. I'll see if I can come up with something clearer.

Reply:
Ah, wait, you edited! Yeah `dwp_override_non_passported` is good, I'll go with that.